tests/components/legrand_rflc/emulation.py

```diff
@@ -0,0 +1,110 @@
+"""Emulation of Legrand RFLC LC7001."""
+import asyncio
+from typing import Final
+
+from homeassistant.components.legrand_rflc.const import DOMAIN
+from homeassistant.const import CONF_AUTHENTICATION, CONF_HOST, CONF_PORT
+
+from tests.common import MockConfigEntry
+
+
+class Server:
```

Comment:
Why do we need to setup an actual server, can't we mock out the relevant parts of the lib instead of this?

Reply:
Emulating a server is a more complete test because it tests the underlying library and communication protocol. I suppose that mocking out library calls would work for just testing the HA usage but (as the library author too) I would prefer to keep the more complete test.